### PR TITLE
Securing Django API

### PR DESCRIPTION
**Pull request by paulodhiambo**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by paulodhiambo** [comment]
Closes #701

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

File: articles/securing-django-api/index.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+ We need to create a user and test the token generation if it works as expected.
+ To create a user for testing, we use a `manage.py` command-line utility.
+ Run the command:
+ ```bash
+ python manage.py createsuperuser --username paul --email paul@gmail.com
+ ```
+### Project setup
+
+Since this is a continuation of our previous article, We will be using the `Django_todo` application that we created previously.
+If you don't have the application we created in the previous article, you can clone it from [here]().
```
Comment (lalith1403) [inline articles/securing-django-api/index.md:44]: Attach the GitHub link! I will review it once the link is inserted.

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/securing-django-api/index.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    'django.contrib.staticfiles',
+    'todo',
+    'rest_framework',
+    'rest_framework.authtoken'
+]
+```
+Add `'rest_framework.authentication.TokenAuthentication'` to `REST_FRAMEWORK` dictionary in the `django_todo` project `settings.py`.
+
+```python
+REST_FRAMEWORK = {
```
Comment (lalith1403) [inline articles/securing-django-api/index.md:24]: Check this section. There is some overlap with the material presented later. In the later section, you mention the same set of steps, but with the addition of coreapi, which is not mentioned here.

File: articles/securing-django-api/index.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+### User token endpoint
+Django rest framework comes with an endpoint that users can use to generate their authentication tokens by providing their valid username and password.
+
+In the `django_todo` project directory add the api endpoint for token generation in the `urls.py` file.
+```python
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('api/v1/todo/', include("todo.urls")),
+    path('docs/', include_docs_urls(title='Todo Api')),
+    path('api/token', obtain_auth_token, name="auth_toke"),#token generation url
```
Comment (lalith1403) [inline articles/securing-django-api/index.md:149]: where have you defined obtain_auth_token? I am getting an error at this step! Can you check this out?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Hey, the topic is fascinating and useful. However, the presentation needs a lot of work. You begin with the previous article. But, in this article, you mention `api/token` in the urls.py file. But in the previous article, there is no mention of the same. So I suggest doing the following. Start by cloning the repo that you have given the link to. Once done, try following the commands given in this article. And then explain the steps accordingly. 

Also, mention the URLs to be used to perform GET and POST requests on POSTMAN. 

I am enjoying implementing the code given in this article. You explain well. 👍. Ping me on Slack if any doubts regarding the same.

File: articles/securing-django-api/index.md (hunk 2 above)
Comment (paulodhiambo) [reply articles/securing-django-api/index.md:28]: I have corrected this. The coreapi i explained about it in the previous article.

**Comment by lalith1403** [comment]
Perfect. The code works! Good job :)

**Comment by hectorkambow** [comment]
Good morning/evening Paul - quick question about your PR - I'm going through it for my final review - but I wanted to make sure the images are in order:
https://github.com/section-io/engineering-education/pull/851
It looked like it was missed one - I added it in - but not sure if in the correct place. @paulodhiambo

**Comment by paulodhiambo** [comment]
@hectorkambow you have added the image in the correct place. Sorry for that oversight error.

**Comment by hectorkambow** [comment]
@paulodhiambo One more clarification: 
image

**Comment by paulodhiambo** [comment]
@hectorkambow I was explaining the part of code snippet that I provided just above that section. 
```python
header = eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9
payload = eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTQzODI4NDMxLCJqdGkiOiI3ZjU5OTdiNzE1MGQ0NjU3OWRjMmI0OTE2NzA5N2U3YiIsInVzZXJfaWQiOjF9
signature = Ju70kdcaHKn1Qaz8H42zrOYk0Jx9kIckTn9Xx7vhikY
```

**Comment by hectorkambow** [comment]
> @hectorkambow I was explaining the part of code snippet that I provided just above that section.
> 
> ```python
> header = eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9
> payload = eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTQzODI4NDMxLCJqdGkiOiI3ZjU5OTdiNzE1MGQ0NjU3OWRjMmI0OTE2NzA5N2U3YiIsInVzZXJfaWQiOjF9
> signature = Ju70kdcaHKn1Qaz8H42zrOYk0Jx9kIckTn9Xx7vhikY
> ```

Okay - cool. I did see that - but just wanted to be sure that it wasn't missing anything. 
Thank you @paulodhiambo  🚀

**Pull request merged**